arangod/Cluster/ClusterTypes.h

```diff
@@ -86,9 +86,51 @@ struct AnalyzersRevision {
   using Revision = uint64_t;
   using Ptr = std::shared_ptr<AnalyzersRevision const>;
 
+
   static constexpr Revision LATEST = std::numeric_limits<uint64_t>::max();
   static constexpr Revision MIN = 0;
 
+  /// @brief Analyzers revisions used in query.
+  /// Stores current database revision
+  /// and _system database revision (analyzers from _system are accessible from other databases)
+  /// If at some point we will decide to allow cross-database anayzer usage this could
+  /// became more complicated. But for now  we keep it simple - store just two members
+  struct QueryAnalyzerRevisions {
+    constexpr QueryAnalyzerRevisions(Revision current, Revision system)
+      : currentDbRevision(current), systemDbRevision(system) {}
+
+    QueryAnalyzerRevisions() = default;
+
+    void toVelocyPack(VPackBuilder& builder) const;
+    bool fromVelocyPack(velocypack::Slice slice, std::string& error);
+
+    bool isDefault() const noexcept {
+      return currentDbRevision == AnalyzersRevision::MIN &&
+             systemDbRevision == AnalyzersRevision::MIN;
+    }
+
+    bool operator==(QueryAnalyzerRevisions const& other) const noexcept {
+      return currentDbRevision == other.currentDbRevision &&
+             systemDbRevision == other.systemDbRevision;
+    }
+
+    std::ostream& print(std::ostream& o) const;
+
+    bool operator!=(QueryAnalyzerRevisions const& other) const noexcept {
+      return !(*this == other);
+    }
+
+    /// @brief Gets analyzers revision to be used with specified database
+    /// @param vocbase database name
+    /// @return analyzers revision
+    Revision getVocbaseRevision(DatabaseID const& vocbase) const noexcept;
+
+    Revision currentDbRevision{ AnalyzersRevision::MIN};
+    Revision systemDbRevision{ AnalyzersRevision::MIN};
```

Comment:
I suggest to mark both members with `const` to emphasize immutability

Reply:
Good point about enforcing immutability. But let`s just make members private (as actually no direct access is needed) and give no setters. Making members const will complicate fromVelocyPack implementation.